### DISCUSSION TITLE
Why does emit_signal still want stringName as argument?

### DISCUSSION THREAD

**Discussion by Overvault-64** [godotengine/godot-proposals#5922] [Scripting]
In Godot 4 we wait for signals directly with `await mynode.signal_name` and the editor auto-completes the identifier. Super comfy.

On the other hand, `emit_signal()` auto-completes while we are typing the identifier like in `emit_signal(signal_name)` BUT then it wants us to add quotes because it only accepts stringName.

Any thoughts?

**Comment by Spartan322** [comment]
Probably because you can just do [`signal_name.emit()`](https://docs.godotengine.org/en/latest/classes/class_signal.html#class-signal-method-emit), so unless you need to emit a signal by name, its safer and easier just to use the signal's emit method. (as the static analyzer for GDScript is certainly better at erroring on erroneous emits like that since you're taking advantage of the type system and variable validation) Its not like emit_signal ever supported describing the object of a signal before so what do you suggest happens if you try to do `emit_signal(other_node.signal_name)`?  (emit_signal still applies to its respective self after all) With StringName this can't be a problem. (even if its easily solved, again might as well do `other_node.signal_name.emit()`)

**Reply by Overvault-64** [reply]
Makes sense
